### ISSUE TITLE
Autocomplete not always working

### ISSUE THREAD

**Issue by Light2Dark** [marimo-team/marimo#2877]
### Describe the bug

Kinda 2 issues I've seen, I think there are some improvements we can make to have the autocomplete powers of a Jupyter notebook.
There are some other issues but I saw on Discord there's some work on a [native vscode plugin](https://discord.com/channels/1059888774789730424/1248732986187448331/1250561528885674014) so I wonder what's the best way forward. Thanks!

1. Doesn't work all the time
![Image](https://github.com/user-attachments/assets/a820f666-58db-4faf-bb2b-346b96d551ff)
This is likely due to `frontend/src/core/codemirror/completion/Autocompleter.ts`
```.ts
  asCompletionResult(
    position: number,
    message: CompletionResultMessage,
  ): CompletionResult {
    console.log("message", message)
    return {
      from: position - message.prefix_length,
      options: message.options.map((option) => {
        return {
          label: option.name,
          type: option.type,
          info: () => constructCompletionInfoNode(option.completion_info),
        };
      }),
      // validFor: /^\w*$/,
    };
  },
```
commenting out the validFor will give proper autocomplete.

2. Not showing docstrings upon '.'
![Image](https://github.com/user-attachments/assets/63f07116-17c2-4320-ac6a-503596c1f1e7)
This is due to docstrings_limit at `marimo/_runtime/complete.py#L316`
```.py
def complete(
    request: CodeCompletionRequest,
    graph: dataflow.DirectedGraph,
    glbls: dict[str, Any],
    glbls_lock: threading.RLock,
    stream: Stream,
    docstrings_limit: int = 80, # Here, configuring it to something higher will help
    timeout: float | None = None,
    prefer_interpreter_completion: bool = False,
) -> None:
```

Similar issue: #2080 

### Environment

```
{
  "marimo": "0.9.20",
  "OS": "Darwin",
  "OS Version": "23.6.0",
  "Processor": "arm",
  "Python Version": "3.12.7",
  "Binaries": {
    "Browser": "131.0.6778.69",
    "Node": "v20.11.0"
  },
  "Dependencies": {
    "click": "8.1.3",
    "docutils": "0.21.2",
    "itsdangerous": "2.2.0",
    "jedi": "0.19.1",
    "markdown": "3.7",
    "narwhals": "1.12.1",
    "packaging": "24.1",
    "psutil": "6.1.0",
    "pygments": "2.18.0",
    "pymdown-extensions": "10.12",
    "pyyaml": "6.0.2",
    "ruff": "0.6.9",
    "starlette": "0.41.2",
    "tomlkit": "0.13.2",
    "typing-extensions": "4.12.2",
    "uvicorn": "0.32.0",
    "websockets": "12.0"
  },
  "Optional Dependencies": {
    "duckdb": "1.1.2",
    "pandas": "2.2.3",
    "polars": "1.12.0",
    "pyarrow": "18.0.0"
  }
}
```

**Comment by mscolnick** [comment]
@Light2Dark im down for both of these changes. did you notice any issues when removing `validFor: /^\w*$/,`? i am find to remove that.

> Not showing docstrings upon '.'

What exactly are you experiencing here? I sometimes do get completions on `.` but do see it not always showing up.

**Comment by Light2Dark** [comment]
> did you notice any issues when removing validFor: /^\w*$/

yes, it's a little hard to debug, but I am finding some (like plotly commands) not autocompleting. Let me investigate, I'm not sure if it's because of this / a different bug.

> What exactly are you experiencing here

I am expecting the docstring to show up as soon as I enter `.` like this:
![Image](https://github.com/user-attachments/assets/748b363f-35f9-4b99-8fef-1eb2ee5678d1)

I see the backend is returning empty completion docstrings when the number of options are too large, due to the `docstrings_limit`.

**Comment by mscolnick** [comment]
Closing as completed by the new [LSP](https://docs.marimo.io/guides/editor_features/language_server/) ([via many PRs](https://github.com/marimo-team/marimo/pulls?q=sort%3Aupdated-desc+is%3Apr+lsp+is%3Aclosed+))

Which can be tested with:
```
uvx --with "marimo[lsp]" --with numpy marimo new
```
